extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -115,11 +112,10 @@ export class AmpAdExit extends AMP.BaseElement {
     if (HostServices.isAvailable(this.getAmpDoc())) {
       HostServices.exitForDoc(this.getAmpDoc())
           .then(exitService => exitService.openUrl(finalUrl))
-          .catch(errorCode => {
+          .catch(error => {
             // TODO: reporting on errors
-            dev().fine(TAG, 'HostServiceError: ' + errorCode);
-            // fallback on browser API on environment miss match
-            if (errorCode === HostServiceError.MISMATCH) {
+            dev().fine(TAG, 'ExitServiceError - fallback=' + error.fallback);
+            if (error.fallback) {
```

Comment:
@choumx since you mentioned in previous comment, can you help confirm if you see this as an issue (error.fallback) with closure compiler? We have the typedef of this error object in host-services.js

Reply:
I don't believe CC type-checks Promise rejections and I'm not sure it's possible since any nested runtime exception can reject the Promise.

So I'd also check that error is not null, e.g. `error && error.fallback`.

Reply:
I tested locally with CC, in both multi-pass and single-pass modes.

multi-pass does not mangle the "fallback" field.
single-pass mangled the "fallback" field to same string. even in the catch block, without an explicit type-casting.

Is CC that smart :-) @erwinmombay any insights?

Reply:
@lannka are you saying it mangled it uniformly across just this same file, or across binaries?

Reply:
Mangling to the same string in both cases sounds like what I would expect.  The single-pass build turns on advanced optimizations, which includes this kind of mangling.

This was also an issue with https://github.com/ampproject/amphtml/pull/20455

Reply:
`foo['bar']` is safe, but `foo.bar` is open for renaming.

Reply:
I expect there are a lot of latent issues in this category, since we haven't previously been writing code while thinking about the changes the compilation-mode ADVANCED allows the compiler to make.

Reply:
@erwinmombay yep, I saw the mangling is consistent across binaries in single-pass.
It's a myth to me how CC did that, as I thought it should have lost the type info about the error object in the catch block. 

It must not be "luck", because I ran single-pass twice, the mangling is different between the runs, but the same within each run.

Reply:
@jeffkaufman yep, I plan to add some integration test later to make sure the host-service error handling works regardless of the compilation mode.

Reply:
I know it can trace to a certain point, but i'd be scared to rely on that behavior without proper typing

Reply:
haha, true. I honestly don't know how smart CC can be. Would an integration test make you comfortable? Or you think we should just make it a JSON object?

Reply:
Or, as you pointed out, force a typecast here